lib/wallaroo/core/topology/router.pony

```diff
@@ -1322,7 +1327,45 @@ class val LocalStatelessPartitionRouter is StatelessPartitionRouter
       end
     end
     LocalStatelessPartitionRouter(_partition_id, _worker_name, _step_ids,
-      consume new_partition_routes)
+      consume new_partition_routes, _steps_per_worker)
+
+  fun calculate_shrink(remaining_workers: Array[String] val):
+    StatelessPartitionRouter
+  =>
+    let new_partition_count = remaining_workers.size() * _steps_per_worker
```

Comment:
i think this needs some explanation of the algo that is being executed. It's hard to follow in the code.

Reply:
Done